### PR TITLE
sessionctx/variable: fix select variable return wrong result when variable is only global scope variable

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve? 
fix issue:https://github.com/pingcap/tidb/issues/8967

### What is changed and how it works?
Always get global value when the variable is only global scope.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Side effects

Related changes

 - Need to cherry-pick to the release branch

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/8968)

### CODE REVIEW

**Comment by crazycs520** [comment]
@XuHuaiyu @zimulala PTAL

File: planner/core/expression_rewriter.go [hunk 1]
```diff
@@ -865,7 +865,13 @@ func (er *expressionRewriter) rewriteVariable(v *ast.VariableExpr) {
 			return
 		}
 	}
-	if v.IsGlobal {
+	sysVar := variable.SysVars[name]
+	if sysVar == nil {
+		er.err = errors.Trace(variable.UnknownSystemVar.GenWithStackByArgs(name))
```
Comment (zz-jason) [inline planner/core/expression_rewriter.go]: no need to call `errors.Trace()` since the error is generated with a call stack.

Comment (XuHuaiyu) [inline planner/core/expression_rewriter.go]: Add a test to cover this check?

Comment (XuHuaiyu) [reply planner/core/expression_rewriter.go]: Please check this comment~

Comment (crazycs520) [reply planner/core/expression_rewriter.go]: done.

**Comment by zz-jason** [comment]
@crazycs520 Should we cherry pick this change to release-2.1 and release-2.0?

**Pull request merged**